tests/test_event_detection.py

```diff
@@ -1,4 +1,117 @@
+    event_band = [15, 29]  # beta band (Hz)
+    thresh_fom = 6.0  # factor-of-the-median threshold
+
+    # get python evs by running find_events()
+    py_ev_dict = list()
+    for _, id_val in enumerate(subj_ids):
+        fname = op.join(
+            root_dir, 'data', 'prestim_humandetection_600hzMEG_subject' +
+            id_val + '.mat')
+        raw_data = loadmat(fname)
+        data = raw_data['prestim_raw_yes_no']
+        # calculate TFR
+        tfrs = se.tfr(data, freqs, samp_freq)
+
+        # find events
+        subj_evs = se.find_events(tfr=tfrs, times=times, freqs=freqs,
+                                  event_band=event_band, thresholds=None,
+                                  threshold_FOM=thresh_fom)
+        py_ev_dict.append(subj_evs)
+
+    # take ev counts and timing latencies from respective dictionaries
+    # and structures to put into a subj x trial matrix for comparison
+    py_ev_count_mat = np.zeros((10, 200))
+    for subj_idx, id_val in enumerate(subj_ids):
+        py_ev_count_mat[subj_idx, :] = [
+            len(trial) for trial in py_ev_dict[subj_idx]]
+
+    # matlab - these are already in an array, so just reshape
+    matlab_ev_count_mat = [
+        trial.shape[1] for trial in matlab_ev_struct['event_times'][:, 0]]
+    matlab_ev_count_mat = np.reshape(matlab_ev_count_mat, (10, 200))
+
+    # Overall check that same # of evs detected
+    # (or a tolerable # of differences)
+    assert np.mean(np.abs(
+        matlab_ev_count_mat - py_ev_count_mat
+        )) <= dev_count_avg, "Should be mostly same number of evs"
+
+    # Once assured number of evs are within tolerable limits, extract evs
+    # matlab evs
+    same_ev_count_bool = (matlab_ev_count_mat == py_ev_count_mat)
+    same_ev_count_bool_reshape = np.reshape(same_ev_count_bool, (2000,))
+
+    # get event timing in trials in which the event count was equal
+    matlab_ev_timing_list = list()
+    py_ev_timing_list = list()
+    # extract latencies from matlab structure, which has all subject trials
+    # in array of arrays within dictionary key
+    # first, extract the events
+    trial_indices = np.where(same_ev_count_bool_reshape)[0]
+    event_times = [
+        matlab_ev_struct['event_times'][trial, 0].flatten()
+        for trial in trial_indices
+    ]
+
+    # then, concatenate the event times into a single list
+    matlab_ev_timing_list = np.concatenate(event_times)
+
+    # extract latencies from py dictionaries for each subject
+    # First, create a list of sorted event times for each trial
+    trial_event_times = []
+    for subj_idx, subj_evs in enumerate(py_ev_dict):
+        for trial_idx in range(200):
+            if same_ev_count_bool[subj_idx, trial_idx]:
+                event_times = [event['Peak Time']
+                               for event in subj_evs[trial_idx]]
+                sorted_event_times = np.sort(event_times)
+                trial_event_times.append(sorted_event_times)
```

Comment:
This works! My last recommendation would be to define the `n_subjs = 10` and `n_trials = 200` at the top of the function and then reference those values throughout the function instead of hardcoding these numbers each time. This way it's 1) easier for a new contributor to read the code and understand exactly why e.g. you're reshaping an array to certain size and 2) simpler to debug/modify if the underlying code format should change.